src/tigerbeetle.zig

```diff
@@ -248,6 +248,41 @@ pub const CreateTransfersResult = extern struct {
     }
 };
 
+pub const GetAccountTransfers = extern struct {
+    /// The account id.
+    account_id: u128,
+
+    /// Use this field for pagination, transfers will be returned from this timestamp
+    /// depending on the sort order. May be zero.
+    timestamp: u64 = 0,
+
+    /// Maximum number of transfers that can be returned by this query.
+    /// Use zero for no limit (limited only by the maximum response size).
+    limit: u32 = 0,
+
+    flags: GetAccountTransfersFlags = .{},
+
+    comptime {
+        assert(@sizeOf(GetAccountTransfers) == 32);
+        assert(stdx.no_padding(GetAccountTransfers));
+    }
+};
+
+pub const GetAccountTransfersFlags = packed struct(u32) {
+    /// Whether to include debit transfers where `debit_account_id` matches.
+    debits: bool = true,
+    /// Whether to include credit transfers where `credit_account_id` matches.
+    credits: bool = true,
+    /// Results are sorted by timestamp, in ascending or descending order.
+    order_by_descending: bool = false,
```

Comment:
:thinking: What do you think of `order: enum(u1) { ascending, descending } = .ascending,`?
(Extending on up to the language clients' APIs, even.)

It's not a pattern we use for any other flags, and it would require some work on the client code generation side...
Then again this is a temporary API, so maybe not worth it. But we could consider it for the full query API.

Reply:
Yes, I tried that, but it seemed to be a footgun, especially when one wanted to toggle the flag with `.ascending` 😭

I'm not happy with those flags, as a suggestion, we could use two different enums:

```zig
pub const GetAccountTransfers = extern struct {
    /// The account id.
    account_id: u128,

    /// Use this field for pagination, transfers will be returned from this timestamp
    /// depending on the sort order. May be zero.
    timestamp: u64 = 0,

    /// Maximum number of transfers that can be returned by this query.
    /// Use zero for no limit (limited only by the maximum response size).
    limit: u32 = 0,

    where: enum(u8) { debits_and_credits = 0, credits = 1, debits = 2 },

    order_by: enum(u8) { ascending = 0, descending = 1 },

    reserved: u16 = 0,

    comptime {
        assert(@sizeOf(GetAccountTransfers) == 32);
        assert(stdx.no_padding(GetAccountTransfers));
    }
};
```

Reply:
> especially when one wanted to toggle the flag with `.ascending`

Can you elaborate; I'm not sure what you mean?

Reply:
> Can you elaborate; I'm not sure what you mean?

The Zig side is fine, as it uses a `packed` struct representing each bit. But as a public API it's not clear to the user that the enum `.ascending` means "remove the flag" and `.descending` "add the flag".

Some code like this one would be required for all clients:

```zig
// Toggle the order_by flag:
fn setOrderBy(filter: *Self, direction: Direction) void {
    const flag_bit = 1 << 2;
    switch (direction) {
        .ascending => filter.flags &= ~flag_bit,
        .descending => filter.flags |= flag_bit,
    }
}
```

Because of that, I opted to make the flag's name explicit as `order_by_descending`.

Reply:
In a follow-up PR, I'll refactor the Node and DotNet clients not to use bitwise flags (e.g. `flags = A | B`), but instead a helper struct with fields.

The Go and Java clients are already ok.